libknet/threads_send_recv.c

```diff
@@ -469,6 +470,43 @@ int knet_send_sync(knet_handle_t knet_h, const char *buff, const size_t buff_len
 	return err;
 }
 
+
+static void _close_socket(knet_handle_t knet_h, int sockfd)
+{
+	struct epoll_event ev;
+	int i;
+
+	log_err(knet_h, KNET_SUB_LINK_T, "EOF received on socket fd %d", sockfd);
+
+	memset(&ev, 0, sizeof(struct epoll_event));
+
+	ev.events = EPOLLIN;
+	ev.data.fd = sockfd;
+	if (epoll_ctl(knet_h->recv_from_links_epollfd, EPOLL_CTL_DEL, sockfd, &ev)) {
+		log_err(knet_h, KNET_SUB_LISTENER, "Unable to remove EOFed socket from epoll pool: %s",
+			strerror(errno));
+	}
+
+	/* Tell transport that the FD has been closed */
+	for (i=0; i<KNET_MAX_TRANSPORTS; i++) {
+		if (knet_h->transports[i] &&
+		    !knet_h->transport_ops[i]->handle_fd_eof(knet_h, sockfd))
+			break;
+	}
+}
+
+static void _handle_socket_notification(knet_handle_t knet_h, int sockfd, struct iovec *iov, size_t iovlen)
+{
+	int i;
+
+	/* Find the transport and post the message */
+	for (i=0; i<KNET_MAX_TRANSPORTS; i++) {
+		if (knet_h->transports[i] && knet_h->transport_ops[i]->handle_fd_notification &&
+		    knet_h->transport_ops[i]->handle_fd_notification(knet_h, sockfd, iov, iovlen))
+			break;
+	}
+}
+
```

Comment:
Those 2 functions should probably move to transports_common.c. thread_send_recv is getting too big (planning to split it in send and recv and shouldn´t do socket management.